src/uu/cksum/src/cksum.rs

```diff
@@ -367,6 +369,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         None
     };
 
+    if algo_name == "bsd" && check {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "--check is not supported with --algorithm={bsd,sysv,crc}",
```

Comment:
The `if`-condition and the error message disagree on the set of supported algorithms. Huh?

Also, I suggest changing the commit message to something more concrete like:
`cksum: handle bsd/sysv/crc check incompatibilities like GNU`

Reply:
> The if-condition and the error message disagree on the set of supported algorithms. Huh?

nope, read again :)

Reply:
```
    if algo_name == "bsd" && check {
        return Err(io::Error::new(
            io::ErrorKind::InvalidInput,
            "--check is not supported with --algorithm={bsd,sysv,crc}",
```

The error message says that (among other things) sysv is not supported with `--check`. The `if`-condition wouldn't trigger in that case. What am I overlooking?